scripts/core/runtime.yml

```diff
@@ -64,3 +174,184 @@ params:
+      desc: |
+            [out][optional] returns the total number of adapters available. 
+returns:
+    - $X_RESULT_ERROR_INVALID_SIZE
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Releases the adapter handle reference indicating end of its usage"
+class: $x
+name: AdapterRelease
+decl: static
+ordinal: "3"
+details:
+    - "When the reference count of the adapter reaches zero, the adapter may perform adapter-specififc resource teardown"
+params:
+  - type: "$x_adapter_handle_t"
+    name: hAdapter
+    desc: |
+          [in] Adapter handle to release
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get a reference to the adapter handle."
+class: $x
+name: AdapterRetain
+decl: static
+ordinal: "4"
+details:
+    - "Get a reference to the adapter handle. Increment its reference count"
+params:
+  - type: "$x_adapter_handle_t"
+    name: hAdapter
+    desc: |
+          [in] Adapter handle to retain
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get the last adapter specific error."
+details: |
+    To be used after another entry-point has returned
+    $X_RESULT_ERROR_ADAPTER_SPECIFIC in order to retrieve a message describing
+    the circumstances of the underlying driver error and the error code
+    returned by the failed driver entry-point.
+
+    * Implementations *must* store the message and error code in thread-local
+      storage prior to returning $X_RESULT_ERROR_ADAPTER_SPECIFIC.
+
+    * The message and error code storage is will only be valid if a previously
+      called entry-point returned $X_RESULT_ERROR_ADAPTER_SPECIFIC.
+
+    * The memory pointed to by the C string returned in `ppMessage` is owned by
+      the adapter and *must* be null terminated.
+
+    * The application *may* call this function from simultaneous threads.
+
+    * The implementation of this function *should* be lock-free.
+
+    Example usage:
+
+    ```cpp
+    if ($xQueueCreate(hContext, hDevice, nullptr, &hQueue) ==
+            $X_RESULT_ERROR_ADAPTER_SPECIFIC) {
+        const char* pMessage;
+        int32_t error;
+        $xAdapterGetLastError(hAdapter, &pMessage, &error);
+    }
+    ```
+class: $x
+name: AdapterGetLastError
+decl: static
+ordinal: "5"
+params:
+    - type: $x_adapter_handle_t
+      name: hAdapter
+      desc: "[in] handle of the adapter instance"
+    - type: const char**
+      name: ppMessage
+      desc: >
+          [out] pointer to a C string where the adapter specific error message
+          will be stored.
+    - type: int32_t*
+      name: pError
+      desc: >
+          [out] pointer to an integer where the adapter specific error code
+          will be stored.
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Supported adapter info"
+class: $x
+name: $x_adapter_info_t
+typed_etors: True
+etors:
```

Comment:
`REFERENCE_COUNT` and a test ?

Reply:
Added `REFERENCE_COUNT` and tests for this and all the other new entry points